File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.java
class Solution {
    public int[] answerQueries(int[] nums, int[] queries) {
         int N=nums.length;
        int M=queries.length;
        
        int ans[] = new int[M];
         Arrays.sort(nums);
        
        for(int i=0;i<M;i++){
            int sum=0;
            int count=0;
            for(int j=0;j<N;j++){
                if(sum+nums[j]<=queries[i]){
                    sum+=nums[j];
                    count++;
                }
            }
            ans[i]=count;
        }
        return ans;
    }
}

File: 2389-longest-subsequence-with-limited-sum/2389-longest-subsequence-with-limited-sum.py
class Solution:
    def answerQueries(self, nums: List[int], queries: List[int]) -> List[int]:
        M,N=len(queries),len(nums)
        nums.sort()
        ans=[]
        def BS(row,t):
            l,r=0,len(row)-1
            while l<=r:
                m=(l+r)//2
                if row[m]<=t:
                    l=m+1
                else:
                    r=m-1 
            return l
        pf=list(accumulate(nums))
        print(pf) 
        for x in queries:
            xx=BS(pf,x)
            ans.append(xx)
        return ans
            
            